ls1intum/Artemis
# Exam exercise submission exam components (text & modeling)

### Checklist

### Motivation and Context


As a lot of functionality is not used for the exam mode, we decided to create an own component for the participation of text and modeling exercises for the exam mode.

### Description

- Removed unnecessary functions and DOM elements for text and modeling exercises
- ParticipationId is now an input 
- The participate exam components will not have the updated version of the layout from https://github.com/ls1intum/Artemis/pull/1653. This will be added after https://github.com/ls1intum/Artemis/pull/1653 was merged in a follow up PR

### Steps for Testing

**Can only be tested locally until student exam participations are loaded into the exam view**

Setup: 
You need to give the new components the participationId as input, therefore you need to create new participation for each exercise in order to test.

- Add `<jhi-text-editor-exam>` to `exam-participation.component.html` and hand over the `participationId` of your newly created participation. You should be able to use the text editor as always.

- Add `<jhi-modeling-submission-exam>` to `exam-participation.component.html` and hand over the `participationId` of your newly created participation. Use `this.modelingSubmissionService.getLatestSubmissionForModelingEditor` instead of the `ExamParticipationService`. You should be able to use modeling editor as always.

1. Log in to Artemis
2. Navigate to /courses/{courseId}/exams/{examsId}
3. Test the components, they should work as always

This should be re-tested thoroughly in the future.

### Screenshots



**NOTE: This is still the "old" layout, this will be updated in a follow up PR**

Text Exercise:
<img width="1275" alt="image" src="https://user-images.githubusercontent.com/41108487/85181280-c8f33d00-b285-11ea-8d1c-5c0ab55d8dcc.png">

Modeling Exercise:
<img width="1251" alt="image" src="https://user-images.githubusercontent.com/41108487/85182048-d3163b00-b287-11ea-88da-4e88671d6d4a.png">
